lib/models/container-store.js

```diff
@@ -75,10 +99,57 @@ ContainerStore.prototype.listContainers = function () {
           Id: container.Id,
           Image: container.Image,
           Created: container.Created,
-          Names: Names
+          Names: Names,
+          Labels: keypather.get(container, 'Config.Labels'),
+          State: container.State
         }
       }.bind(this))
     })
+    .then(function (containers) {
+      return ContainerStore._runFilters(containers, filters)
+    })
+}
+
+ContainerStore._runFilters = function (containers, filters) {
+  Object.keys(filters).forEach(function (filter) {
+    switch (filter) {
+      case 'label':
+        var key = Object.keys(filters[filter])[0]
+        var value = filters[filter][key]
+        containers = containers.filter(filterLabel(key, value))
+        break
+      case 'status':
+        var running = filters[filter] === 'running'
+        containers = containers.filter(filterStatus(running))
+        break
```

Comment:
passing an invalid query parameter isn't a `400` error, it's just not handled (at least, that's my understanding).